server/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsGroup.java

```diff
@@ -20,17 +20,43 @@
 package org.elasticsearch.action.admin.cluster.shards;
 
 import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.search.internal.AliasFilter;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
 
 public class ClusterSearchShardsGroup implements Streamable, ToXContentObject {
 
+    private static final ParseField ALIASES = new ParseField("aliases");
+    private static final ParseField FILTER = new ParseField("filter");
+
+    private static final ConstructingObjectParser<AliasFilter, String> PARSER =
+        new ConstructingObjectParser<>("alias_filter", false, a -> new AliasFilter((QueryBuilder)a[1], (String[])a[0]));
```

Comment:
I think it's really scary to do this backwards for `a[1]` and then `a[0]` here, I think it'd be better to declare the parser fields in the order they are in the constructor